src/generator/operations.ts

```diff
@@ -196,54 +199,41 @@ function formatHeaderResponseValue(propName: string, header: string, schema: Sch
   return text;
 }
 
-function generateMultiRespComment(op: Operation): string {
-  const returnTypes = new Array<string>();
-  for (const response of values(op.responses)) {
-    // the operation might contain a mix of schemas and non-schema responses
-    if (isSchemaResponse(response)) {
-      returnTypes.push(`*${response.schema.language.go!.responseType.name}`);
-    }
-  }
-  return `// Possible return types are ${returnTypes.join(', ')}\n`;
-}
-
 function getZeroReturnValue(op: Operation, apiType: 'api' | 'op' | 'handler'): string {
-  if (!op.responses) {
-    // no responses return *http.Response
-    return 'nil';
-  }
-  if (isMultiRespOperation(op)) {
-    // multi-response APIs return interface{}
-    return 'nil';
-  }
-  const schemaResponse = getResponse(op);
   let returnType = 'nil';
   if (isLROOperation(op)) {
     if (apiType === 'op') {
       // the operation returns an *azcore.Response
       returnType = 'nil';
     } else if (apiType === 'api') {
-      returnType = 'HTTPPollerResponse{}';
-      if (schemaResponse !== undefined) {
-        returnType = `${schemaResponse.schema.language.go!.lroResponseType.language.go!.name}{}`;
-      }
+      returnType = `${getResponseEnvelopeName(op)}{}`;
     } else {
-      returnType = 'nil';
-      if (schemaResponse !== undefined) {
-        returnType = `${schemaResponse.schema.language.go!.responseType.name}{}`;
+      if (isPageableOperation(op)) {
+        returnType = `${getFinalResponseEnvelopeName(op)}{}`;
+      } else if (getSchemaResponse(op) !== undefined) {
+        returnType = `${getResponseEnvelopeName(op)}{}`;
       }
     }
-  } else if (schemaResponse !== undefined) {
-    // simple schema response
-    returnType = `${schemaResponse.schema.language.go!.responseType.name}{}`;
-  } else if (op.language.go!.headAsBoolean === true) {
-    // NOTE: this case must come after the hasSchemaResponse() check to properly handle
-    //       the intersection of head-as-boolean with modeled response headers
-    returnType = 'BooleanResponse{}';
+  } else {
+    returnType = `${getResponseEnvelopeName(op)}{}`;
```

Comment:
Just wondering if this conditional can be modified since I see three conditions returning the same call to `getResponseEnvelopeName`?

Reply:
Good catch, done